src/main/fc/rc.c

```diff
@@ -489,141 +492,177 @@ static FAST_CODE void processRcSmoothingFilter(void)
 
-    const float rcCommandDeltaAbs = fabsf(rcCommand[axis] - prevRcCommand[axis]);
+    const float rcCommandDelta = rcCommand[axis] - prevRcCommand[axis];
     prevRcCommand[axis] = rcCommand[axis];
+    float rcCommandDeltaAbs = fabsf(rcCommandDelta);
 
-    float setpoint = rawSetpoint[axis];
-    float setpointSpeed = (setpoint - prevSetpoint[axis]);
+    const float setpoint = rawSetpoint[axis];
+    const float setpointDelta = setpoint - prevSetpoint[axis];
     prevSetpoint[axis] = setpoint;
-    float setpointAcceleration = 0.0f;
-
-
-    if (axis == FD_ROLL) {
-        DEBUG_SET(DEBUG_FEEDFORWARD, 0, lrintf(setpoint * 10.0f)); // un-smoothed final feedforward
-    }
-
-    // attenuators
-    float zeroTheAcceleration = 1.0f;
-    float jitterAttenuator = 1.0f;
-    if (pid->feedforwardJitterFactor) {
-        if (rcCommandDeltaAbs < pid->feedforwardJitterFactor) {
-            jitterAttenuator = MAX(1.0f - (rcCommandDeltaAbs + prevRcCommandDeltaAbs[axis]) * pid->feedforwardJitterFactorInv, 0.0f);
-            // note that feedforwardJitterFactorInv includes a divide by 2 to average the two previous rcCommandDeltaAbs values
-            jitterAttenuator = 1.0f - jitterAttenuator * jitterAttenuator;
-        }
-    }
-    prevRcCommandDeltaAbs[axis] = rcCommandDeltaAbs;
 
-    // interpolate setpoint if necessary
-    if (rcCommandDeltaAbs) {
-        // movement!
-        if (prevDuplicatePacket[axis] == true) {
-            rxRate = 1.0f / (rxInterval + prevRxInterval);
-            zeroTheAcceleration = 0.0f;
-            // don't add acceleration, empirically seems better on FrSky
-        }
-        setpointSpeed *= rxRate;
-        prevDuplicatePacket[axis] = false;
-    } else {
-        // no movement!
-        if (prevDuplicatePacket[axis] == false) {
-            // first duplicate after movement
-            setpointSpeed = prevSetpointSpeed[axis];
-            if (fabsf(setpoint) < 0.95f * maxRcRate[axis]) {
-                setpointSpeed += prevAcceleration[axis];
+    float setpointSpeed = 0.0f;
+    float setpointSpeedDelta = 0.0f;
+    float feedforward = 0.0f;
+
+    if (pid->feedforwardInterpolate) {
+        static float prevRxInterval;
+        // for Rx links which send duplicate data packets when they send telemetry
+        // extrapolate setpointSpeed when a duplicate is detected, to minimise steps in feedforward
+        const bool isDuplicate = rcCommandDeltaAbs == 0;
+        if (!isDuplicate) {
+            // movement!
+            // but, if the packet before this was also a duplicate,
+            // calculate setpointSpeed over the last two intervals
+            if (isPrevPacketDuplicate[axis]) {
+                rxRate = 1.0f / (rxInterval + prevRxInterval);
             }
-            zeroTheAcceleration = 0.0f; // force acceleration to zero
+            setpointSpeed = setpointDelta * rxRate;
+            isPrevPacketDuplicate[axis] = false;
```

Comment:
IMO `isPrevPacketDuplicate[axis] = isDuplicate;` after if(!isDuplicate)` is slightly more obvious.

Reply:
Maybe :-) I kind of like true and false, but will use isDuplicate.